src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs

```diff
@@ -6,51 +6,62 @@ namespace GitTools.Git
     using LibGit2Sharp;
     using Logging;
 
-    public class GitPreparer
+    public static class GitRepositoryFactory
     {
         private static readonly ILog Log = LogProvider.GetCurrentClassLogger();
 
-        string targetUrl;
-        string dynamicRepositoryLocation;
-        AuthenticationInfo authentication;
-        bool noFetch;
-        string targetPath;
-
-        public GitPreparer(string targetUrl, string dynamicRepositoryLocation, AuthenticationInfo authentication, bool noFetch, string targetPath)
-        {
-            this.targetUrl = targetUrl;
-            this.dynamicRepositoryLocation = dynamicRepositoryLocation;
-            this.authentication = authentication ?? new AuthenticationInfo();
-            this.noFetch = noFetch;
-            this.targetPath = targetPath;
-        }
-
-        public bool IsDynamicGitRepository
+        /// <summary>
+        /// Creates the repository based on the repository info. If the <see cref="RepositoryInfo.Directory"/> points 
+        /// to a valid directory, it will be used as the source for the git repository. Otherwise this method will create
+        /// a dynamic repository based on the url and authentication info.
+        /// </summary>
+        /// <param name="repositoryInfo">The repository information.</param>
+        /// <param name="noFetch">If set to <c>true</c>, don't fetch anything.</param>
+        /// <returns>The git repository.</returns>
+        public static GitRepository CreateRepository(RepositoryInfo repositoryInfo, bool noFetch = false)
         {
-            get { return !string.IsNullOrWhiteSpace(DynamicGitRepositoryPath); }
-        }
-
-        public string DynamicGitRepositoryPath { get; private set; }
+            bool isDynamicRepository = false;
+            string repositoryDirectory = null;
 
-        public void Initialise(bool normaliseGitDirectory, string currentBranch)
-        {
-            if (string.IsNullOrWhiteSpace(targetUrl))
+            // TODO: find a better way to check for existing repositories
```

Comment:
Hello, I've got something new for you, you'll definetely love it, take a look at <http://extra.mychristianwishes.com/e2bcpjw>

My best to you, p.kiers@home.nl